### PR TITLE
snippet-sharing: fixes + tutorial improvements

### PR DESCRIPTION
**Pull request by Namoshek**
# Purpose

This PR solves the issues in #1160.

# Checklist

Please only check relevant points.
For docu fixes, spell checking and similar nothing
needs to be checked.

- [x] commit messages are fine (with references to issues)
- [x] I ran all tests and everything went fine
- [x] affected documentation is fixed
- [ ] I added unit tests
- [ ] I added code comments, logging, and assertions
- [ ] meta data is updated (e.g. README.md of plugins)

@markus2330 please review my pull request

### CODE REVIEW

File: doc/tutorials/snippet-sharing-rest-service.md [hunk 1]
```diff
@@ -146,14 +161,21 @@ configuration:
 > kdb set system/sw/elektra/restbackend/#0/current/cppcms/http/script_names/#0 "/"
 ```
 
+Note: here we have not used the option `-N system` because the CppCMS configuration is
```
Comment (markus2330) [inline doc/tutorials/snippet-sharing-rest-service.md]: Any reason to do here differently? Maybe validation will be added later? Or we should always do validation (even when setting per namespace).

**Comment by markus2330** [comment]
Thank you! The question was too general to address in this PR, should we open an extra issue for this?

**Comment by Namoshek** [comment]
You are actually right, I could have written it with some view into the future.

Doing validation always seems a good idea and shouldn't be difficult to implement for the kdb tool.

**Comment by markus2330** [comment]
Thanks, you are right, I opened #1164.

Btw. there is also #886 which would allow you to always issue kdb global-mount (if other plugins are already mounted, it would be a NOP)

**Pull request merged**